src/sentry/integrations/github/client.py

```diff
@@ -415,13 +420,15 @@ def get_token(self, force_refresh: bool = False) -> str:
 
         token: str | None = self.integration.metadata.get("access_token")
         expires_at: str | None = self.integration.metadata.get("expires_at")
+        now = datetime.utcnow()
 
         if (
             not token
             or not expires_at
-            or (datetime.strptime(expires_at, "%Y-%m-%dT%H:%M:%S") < datetime.utcnow())
+            or (datetime.strptime(expires_at, "%Y-%m-%dT%H:%M:%S") < now)
             or force_refresh
         ):
+            logger.info(f"Token to be refreshed (expires: {expires_at}, now: {now}).")
```

Comment:
I'm suspicious that the issue may happen during the time that the token is being refreshed.
Let's see if this logging would help.